test/integration/progressUpdates.test.js

```diff
@@ -0,0 +1,204 @@
+          if (err) return done(err);
+          expect(res).to.have.status(401);
+          expect(res.body.message).to.be.equal("Unauthenticated User");
+          return done();
+        });
+    });
+
+    it("Returns 404 for invalid task id's", function (done) {
+      chai
+        .request(app)
+        .post(`/progressupdates/123`)
+        .set("Cookie", `${cookieName}=${jwt}`)
+        .send({
+          monitor: false,
+        })
+        .end((err, res) => {
+          if (err) return done(err);
+          expect(res).to.have.status(404);
+          expect(res.body.message).to.be.equal("Task id not found");
+          return done();
+        });
+    });
+  });
+
+  describe("Test the save api", function () {
+    it("Saves the progress updates", function (done) {
+      chai
+        .request(app)
+        .post(`progressupdates/save/${taskId}`)
+        .set("Cookie", `${cookieName}=${jwt}`)
+        .send(progressData[0])
+        .end((err, res) => {
+          if (err) return done(err);
+          expect(res).to.have.status(200);
+          expect(res.body.message).to.be.equal("task update added successfully");
+          expect(res.body.id).to.be.equal(taskId);
+          return done();
+        });
+    });
+
+    it("Gives 400 for invalid request body", function (done) {
+      chai
+        .request(app)
+        .post(`progressupdates/save/${taskId}`)
+        .set("Cookie", `${cookieName}=${jwt}`)
+        .send(progressData[1])
+        .end((err, res) => {
+          if (err) return done(err);
+          expect(res).to.have.status(400);
+          expect(res.body.message).to.be.equal("timestamp is required");
+          return done();
+        });
+    });
+
+    it("Gives 401 for unauthenticated user", function (done) {
+      chai
+        .request(app)
+        .post(`progressupdates/save/${taskId}`)
+        .send(progressData[1])
+        .end((err, res) => {
+          if (err) return done(err);
+          expect(res).to.have.status(401);
+          expect(res.body.message).to.be.equal("Unauthenticated user");
+          return done();
+        });
+    });
+
+    it("Gives 404 for invalid task id", function (done) {
+      chai
+        .request(app)
+        .post(`progressupdates/save/123`)
+        .set("Cookie", `${cookieName}=${jwt}`)
+        .send(progressData[0])
+        .end((err, res) => {
+          if (err) return done(err);
+          expect(res).to.have.status(404);
+          expect(res.body.message).to.be.equal("task id not found");
+          return done();
+        });
+    });
+  });
+
+  describe("Test the get Updates API", function () {
+    it("Returns the latest progress detail", function (done) {
+      chai
+        .request(app)
+        .get(`/progressupdates/${taskId}`)
+        .end((err, res) => {
+          if (err) return done(err);
+          expect(res).to.have.status(200);
+          expect(res.body).to.have.keys(["timestamp", "progress", "plan", "blockers"]);
```

Comment:
res.body will be an array right. In the array each item will have these properties. correct me if I am wrong.